### ISSUE TITLE
org-insert-link-global doesn't work with `global-orglink-mode`

### ISSUE THREAD

**Issue by mohkale** [tarsius/orglink#12]
I've found several other functions are also failing, and can't really work out how I'm expected to hook orglink mode since if I hook it into text-mode it gets called in org-mode and then throws an error.

```lisp
Debugger entered--Lisp error: (error "Orglink Mode doesn’t make sense in Org Mode")
  signal(error ("Orglink Mode doesn’t make sense in Org Mode"))
  error("Orglink Mode doesn't make sense in Org Mode")
  orglink-mode(1)
  turn-on-orglink-mode-if-desired()
  global-orglink-mode-enable-in-buffers()
  run-hooks(after-change-major-mode-hook)
  run-mode-hooks(org-mode-hook)
  org-mode()
  org-get-local-variables()
  org-run-like-in-org-mode(org-insert-link)
  org-insert-link-global()
  funcall-interactively(org-insert-link-global)
  call-interactively(org-insert-link-global nil nil)
  command-execute(org-insert-link-global)
```

**Note**: My value of `orglink-activate-in-modes` is  `'(hledger-mode text-mode prog-mode conf-mode)`.

**Comment by mohkale** [comment]
This seems to fix it.

```diff
diff -u --label /home/mohkale/.cache/emacs/straight/repos/orglink/orglink.el --label \#\<buffer\ orglink.el\> /home/mohkale/.cache/emacs/straight/repos/orglink/orglink.el /tmp/buffer-content-3IfIJV
--- /home/mohkale/.cache/emacs/straight/repos/orglink/orglink.el
+++ #<buffer orglink.el>
@@ -155,7 +155,8 @@
   orglink-mode turn-on-orglink-mode-if-desired)
 
 (defun turn-on-orglink-mode-if-desired ()
-  (when (apply 'derived-mode-p orglink-activate-in-modes)
+  (when (and (not (derived-mode-p 'org-mode))
+             (apply 'derived-mode-p orglink-activate-in-modes))
     (orglink-mode 1)))
 
 (defun orglink-unfontify-region (beg end)
```

**Comment by tarsius** [comment]
I assume that means that you use `org-mode` as `git-commit-major-mode`.

In `org-mode` it doesn't make sense to activate `orglink-mode` and I will adjust `turn-on-orglink-mode-if-desired` and `git-commit-turn-on-orglink` accordingly.

However I can reproduce this issue also when using other modes such as `text-mode`. It appears that the `orglink` implementation has a bug that the `org` implementation doesn't have. I noticed that the `org` implementation has significantly changed since I copied and adjusted it, likely also to fix this bug. I should copy and adjust again to take advantage of the improvements. Doing that might take a little bit longer.
